src/Microsoft.Azure.EventHubs/Amqp/AmqpServiceClient.cs

```diff
@@ -143,6 +144,24 @@ public override Task CloseAsync()
             return this.link.CloseAsync();
         }
 
+        async Task<string> GetTokenString()
+        {
+            using (await this.tokenLock.LockAsync().ConfigureAwait(false))
+            {
+                // Expect maximum 5 minutes of clock skew between client and the service
+                // when checking for token expiry.
+                if (this.token == null || DateTime.UtcNow > this.token.ExpiresAtUtc.Subtract(TimeSpan.FromMinutes(5)))
+                {
+                    var timeoutHelper = new TimeoutHelper(this.eventHubClient.ConnectionStringBuilder.OperationTimeout);
+                    this.token = await this.eventHubClient.TokenProvider.GetTokenAsync(
+                        this.eventHubClient.ConnectionStringBuilder.Endpoint.AbsoluteUri,
+                        ClaimConstants.Manage, timeoutHelper.RemainingTime()).ConfigureAwait(false);
```

Comment:
Good catches Xin. Thx!

---
In reply to: [88131495](https://github.com/Azure/azure-event-hubs-dotnet/pull/65#discussion_r88131495) [](ancestors = 88131495)